src/app/pages/Swap/modals/SwapSettings/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { memo, useCallback, useState } from 'react';
+
+import { isEmpty } from 'lodash';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form-v7';
+
+import AssetField from 'app/atoms/AssetField';
+import { ActionsButtonsBox, PageModal } from 'app/atoms/PageModal';
+import { StyledButton } from 'app/atoms/StyledButton';
+import { ReactComponent as InfoIcon } from 'app/icons/info-icon.svg';
+import { T } from 'lib/i18n';
+
+export type Inputs = {
+  slippageTolerance: number | undefined;
+};
+
+interface SelectTokenModalProps {
+  onSubmit: SubmitHandler<Inputs>;
+  opened: boolean;
+  onRequestClose: EmptyFn;
+}
+
+const SwapSettingsModal = memo<SelectTokenModalProps>(({ onSubmit, opened, onRequestClose }) => {
+  const options = [0.5, 1, 'Custom'] as const;
```

Comment:
Please consider defining the label of 'Custom' option using translation with `t` function in a separate variable and using this value wherever necessary to both enable using translations and prevent typos in code